src/api/metadata/index.ts

```diff
@@ -15,20 +15,29 @@ export class MetadataApi {
     this.context = context
   }
 
-  generateAppDataDoc(metadata: MetadataDoc = {}, optionalProperties?: OptionalAppDataProperties): AppDataDoc {
-    const { appCode = DEFAULT_APP_CODE, environment } = optionalProperties || {}
+  /**
+   * Creates an appDataDoc with the latest version format
+   *
+   * Without params creates a default minimum appData doc
+   * Optionally creates metadata docs
+   */
+  generateAppDataDoc(params?: GenerateAppDataDocParams): LatestAppDataDocVersion {
```

Comment:
WARNING: Interface changed!

Now it's possible to generated the metadata using this method instead of needing to provide the full metadata doc

Reply:
Note that it returns `LatestAppDataDocVersion`, which means it'll always be the latest schema